modules/reindex/src/main/plugin-metadata/plugin-security.policy

```diff
@@ -20,6 +20,14 @@
 grant {
    // reindex opens socket connections using the rest client
    permission java.net.SocketPermission "*", "connect";
-   // rest client uses system properties which gets the default proxy
-   permission java.net.NetPermission "getProxySelector";
+};
+
+grant codeBase "${codebase.elasticsearch-rest-client-6.0.0-beta1-SNAPSHOT.jar}" {
+  // rest client uses system properties which gets the default proxy
+  permission java.net.NetPermission "getProxySelector";
+};
+
+grant codeBase "${codebase.httpasyncclient-4.1.2.jar}" {
+  // rest client uses system properties which gets the default proxy
+  permission java.net.NetPermission "getProxySelector";
```

Comment:
The patch has been incorporated in the upstream project for their next major version (5.0), which combines the async client with the rest of http components.